Split firmware/micropython update, and update the README.md a little, add 1st accelerometer support, mqtt bits - publish on boot, and handle bad mqtt commands

thaytan opened the pull request.

thaytan committed:
software/scripts: Split firmware flash and micropython update

Add a 2nd script to update the micropython code separately from
reflashing the firmware image

thaytan committed:
mqtt: Catch exceptions in message handlers

If a message handler throws an except because of some
malformed input, catch it rather than letting it kill
the main loop and dropping to a REPL.

thaytan committed:
mqtt: Publish to the 'state' topic when booted

Announce our existence to the world, making it easier
to see which bots are alive

thaytan committed:
mpu9250: Convert line endings to unix

thaytan committed:
Create I2C bus in the lolibot module

thaytan committed:
mpu9250: Basic accelerometer support

Initialise the accelerometer, and once per second
output the maximum / minimum reading. Needs hooking
up to MQTT and other stuff

thaytan committed:
mpu9250: Handle module not being installed gracefully

Don't complain or print empty output if the accelerometer
module is not installed

geekscape commented:
Manually "git cherry-pick 8c7de91": README.md: Update with some links to the OHMC2018 wiki

geekscape reviewed (approved):
Expedite with manual cherry-pick

thaytan committed:
Lolibot: Add 'servo' and 'motors' commands

Support directly settings motor PWM controls
motors (1023 0 1023 0) to go forward

Support servo command 'servo [-100..100] for -100
(backward spin) to 100 (forward spin) and 0 for stop

thaytan committed:
lolibot: Check the motors command comes with 4 args

thaytan committed:
udp_control: Add a listener on port 4900 for control

Receive commands with the same format as the MQTT
endpoint directly on port 4900.

Periodically sends a broadcast presence announcement
to the network on that port too

thaytan commented:
I'm not sure what's happening with this pull request. Is it waiting on anything from me to get merged?
